src/converter.py

```diff
@@ -0,0 +1,173 @@
+import datetime
+import json
+
+def convert_tes_to_wrroc(tes_data):
+    try:
+        if not tes_data or not isinstance(tes_data, dict):
+            raise ValueError("Invalid TES data provided")
+
+        if not tes_data.get("executors"):
+            raise ValueError("TES data missing 'executors' list or 'executors' list is empty")
+
+        wrroc = {
+            "@context": "https://w3id.org/ro/crate/1.1/context",
+            "@graph": [
+                {
+                    "@type": "Dataset",
+                    "conformsTo": "https://w3id.org/ro/wfrun/process/0.4"
+                },
+                {
+                    "@type": "CreateAction",
+                    "@id": tes_data.get("id"),
+                    "name": tes_data.get("name", ""),
+                    "description": tes_data.get("description", ""),
+                    "instrument": tes_data["executors"][0].get("image"),
+                    "object": [{"@id": input.get("url"), "name": input.get("path")} for input in tes_data.get("inputs", []) if input.get("url") and input.get("path")],
+                    "result": [{"@id": output.get("url"), "name": output.get("path")} for output in tes_data.get("outputs", []) if output.get("url") and output.get("path")],
+                    "startTime": convert_to_iso8601(tes_data["logs"][0].get("start_time")),
+                    "endTime": convert_to_iso8601(tes_data["logs"][0].get("end_time")),
+                    "actionStatus": convert_status(tes_data.get("state")),
+                },
+                {
+                    "@type": "SoftwareApplication",
+                    "@id": tes_data["executors"][0].get("image"),
+                    "name": tes_data.get("name", "")
+                }
+            ]
+        }
+
+        return json.dumps(wrroc, indent=2)
+    except KeyError as e:
+        raise ValueError(f"Missing expected key: {e}")
+    except IndexError as e:
+        raise ValueError(f"Missing expected index: {e}")
+    except Exception as e:
+        raise ValueError(f"An error occurred during TES to WRROC conversion: {e}")
```

Comment:
None of these are helpful... Think of a proper error handling strategy. Look at other code, read some articles etc.